core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -139,6 +143,9 @@ private Runnable commit(final Substream winningSubstream) {
       if (state.winningSubstream != null) {
         return null;
       }
+      if (retryTracer != null) {
+        retryTracer.commit();
```

Comment:
Can we do this outside (after) the synchronized block? Looks like we can.

Reply:
Done.